t_sparse_list.py

```diff
@@ -300,5 +300,18 @@ def test_remove_default_value_does_nothing(self):
         sl.remove(None)
         self.assertEquals([None, None, None, None], sl)
 
+    def test_set_slice(self):
+        sl = sparse_list.SparseList(4, None)
+        # Only the first two should be assigned.
+        sl[0:2] = [1, 2, 3]
+        self.assertEquals([1, 2, None, None], sl)
+        # Like a normal list, when we assign past the end, it appends.
+        sl[100:] = [4, 5]
+        self.assertEquals([1, 2, None, None, 4, 5], sl)
+        self.assertEquals(len(sl), 6)
```

Comment:
This makes sense, thanks @speedplane.

 I can't find this functionality documented anywhere but both CPython and PyPy behave this way. 

```python
>>> l = [None, None, None, None]
>>> l[100:] = [4, 5]
>>> l
[None, None, None, None, 4, 5]
```

I would have expected an `IndexError` from `list` here because the way this is implemented means that an item can't be retrieved from the index to which it is written:

```python
>>>> l = []
>>>> l[100:] = ['somevalue']
>>>> l[100]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range
```

In https://github.com/johnsyweb/python_sparse_list/commit/828c8bbc9ee3f7d09eab501e9d42146554358e96#diff-e1add0b36b652c44614a561f3bdf4532R121 I made the decision that setting an element past the end of a `SparseList `would behave differently to a `list`, since we set arbitrary elements without having to allocate large blocks of memory.

As I've implemented this `sl[x]` to return the first value allocated by a call to `sl[x:] =`.

If this breaks people's use of `SparseLIst`, then I'd certainly entertain a PR to change the behaviour to match that of `list`. Then I'd have to question whether assigning to a single index past the end of the `SparseLIst` should raise `IndexError` after all.